packages/ckeditor5-ckbox/src/ckboximageedit/ckboximageeditcommand.ts

```diff
@@ -0,0 +1,354 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/* globals document, console, AbortController, URL, window */
+
+/**
+ * @module ckbox/ckboximageedit/ckboximageeditcommand
+ */
+
+import { Command, PendingActions, type Editor } from 'ckeditor5/src/core';
+import { CKEditorError, createElement, retry } from 'ckeditor5/src/utils';
+import type { Element as ModelElement } from 'ckeditor5/src/engine';
+import { Notification } from 'ckeditor5/src/ui';
+import { isEqual } from 'lodash-es';
+
+import CKBoxEditing from '../ckboxediting';
+import { sendHttpRequest } from '../utils';
+import { prepareImageAssetAttributes } from '../ckboxcommand';
+import type { CKBoxRawAssetDefinition, CKBoxRawAssetDataDefinition } from '../ckboxconfig';
+
+import type { ImageUtils } from '@ckeditor/ckeditor5-image';
+
+/**
+ * The CKBox edit image command.
+ *
+ * Opens the CKBox dialog for editing the image.
+ */
+export default class CKBoxImageEditCommand extends Command {
+	/**
+	 * Flag indicating whether the command is active, i.e. dialog is open.
+	 */
+	declare public value: boolean;
+
+	/**
+	 * The DOM element that acts as a mounting point for the CKBox Edit Image dialog.
+	 */
+	private _wrapper: Element | null = null;
+
+	/**
+	 * The states of image processing in progress.
+	 */
+	private _processInProgress = new Map<string, ProcessingState>();
+
+	/**
+	 * @inheritDoc
+	 */
+	constructor( editor: Editor ) {
+		super( editor );
+
+		this.value = false;
+
+		this._prepareListeners();
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public override refresh(): void {
+		const editor = this.editor;
+
+		this.value = this._getValue();
+
+		const selectedElement = editor.model.document.selection.getSelectedElement();
```

Comment:
This should handle selection inside the image caption element.

https://github.com/ckeditor/ckeditor5/blob/490e44e0013a4016601f8484ef27d517d1e71ef0/packages/ckeditor5-image/src/imagetextalternative/imagetextalternativecommand.ts#L30-L31

Reply:
Extracted to follow-up: https://github.com/cksource/ckeditor5-commercial/issues/5741